lib/lhm/chunker.rb

```diff
@@ -48,8 +47,9 @@ def bottom
       @next_to_insert
     end
 
-    def top(stride)
-      [(@next_to_insert + stride - 1), @limit].min
+    def upper_id(next_id, stride)
+      top = connection.select_value("select id from `#{ origin_name }` where id >= #{ next_id } order by id limit 1 offset #{ stride - 1}")
+      [top ? top.to_i : @limit, @limit].min
```

Comment:
BTW: That section of the code is actually doing the same thing than [this method](https://github.com/Shopify/lhm/blob/master/lib/lhm/chunker.rb#L51-L53) in the present incarnation of the code, for the same reasons. 

```
def top(stride)
  [(@next_to_insert + stride - 1), @limit].min
end
```